docs/cart-and-checkout-blocks/faq.md

```diff
@@ -0,0 +1,72 @@
+---
+post_title: Cart and Checkout - Frequently Asked Questions
+menu_title: Extensibility
+tags: reference
+---
+<!-- markdownlint-disable MD041 -->
+
+This document aims to answer some of the frequently asked questions we see from developers extending WooCommerce Blocks.
+
+We will add to the FAQ document as we receive questions, this isn't the document's final form.
+
+If you have questions that aren't addressed here, we invite you to ask them on [GitHub Discussions](https://github.com/woocommerce/woocommerce/discussions) or in the [WooCommerce Community Slack](https://woocommerce.com/community-slack/)
+
+## Cart modifications
+
+### How do I add fees to the cart?
+
+You need to add the fees on the server, this can be achieved using [`extensionCartUpdate`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/docs/third-party-developers/extensibility/rest-api/extend-rest-api-update-cart.md).
+
+This is the server-side code required to add the fee:
+
+```php
+add_action('woocommerce_blocks_loaded', function() {
+  woocommerce_store_api_register_update_callback(
+    [
+      'namespace' => 'extension-unique-namespace',
+      'callback'  => function( $data ) {
+        add_action(
+          'woocommerce_cart_calculate_fees',
+          function( $cart ) {
+            $cart->add_fee( 'Custom Fee', 10 );  
+        } );
+      }
+    ]
+  );
+} );
+```
```

Comment:
This example is incomplete as the fee won't persist on checkout submit flow. Maybe you can change this to something else (apply a coupon programmatically, select a payment method...) anything that actually persists in cart or session.

See https://nadir.blog/2024/07/02/reacting-to-selected-payment-method-in-woocommerce-checkout-block/

Reply:
Ahh, I see what you mean, alright thanks will update

Reply:
The point about fees is that it is an FAQ, specifically for fees, so that's why we want to keep it.

Reply:
Oh, this doesn't need to be an extensionCartUpdate example, you're right.